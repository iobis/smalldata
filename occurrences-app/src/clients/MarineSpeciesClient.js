export function getByName(scientificName) {
  const url = 'http://www.marinespecies.org/rest/AphiaRecordsByName/' + scientificName + '?like=true&marine_only=false'
  return fetch(url).then(response => response.json())
}

export function getById(scientificNameId) {
  const id = scientificNameId.split(':').pop()
  const url = 'http://www.marinespecies.org/rest/AphiaRecordByAphiaID/' + id
  return fetch(url).then(response => response.json())
}

export function isScientificNameId(param) {
  return !!param && param.split(':').length > 0 && /^\d+$/.test(param.split(':').pop())
}
