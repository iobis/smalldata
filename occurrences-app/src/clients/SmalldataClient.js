import { format } from 'date-fns'

export async function getDatasets() {
  const response = await fetch('/api/datasets')
    .then(response => response.json())
  return response.map(renameRefToId)
}

export function renameRefToId({ ref, ...rest }) {
  return ({ id: ref, ...rest })
}

export function datasetTitleOf(dataset) {
  const title = dataset && dataset.title && dataset.title.value
  return title || ''
}

export function getOccurrenceMock() {
  return [{
    id:             1,
    addedDate:      '2011-12-01',
    scientificName: 'Abra Alba',
    dataset:        'NPPSD Short-tailed Albatross Sightings',
    occurrenceDate: '2011-12-09'
  }, {
    id:             2,
    addedDate:      '2011-12-01',
    scientificName: 'Abra Alba',
    dataset:        'NPPSD Short-tailed Albatross Sightings',
    occurrenceDate: '2011-12-09'
  }]
}

export async function postOccurrence({ occurence }) {
  const userRef = 'ovZTtaOJZ98xDDY'
  const occurrence = {
    core:       'occurrence',
    occurrence: [{
      tdwg: {
        datasetID:                     occurence.dataset.id,
        datasetName:                   occurence.dataset.title.value,
        basisOfRecord:                 occurence.occurrenceData.basisOfRecord.charAt(0).toUpperCase() + occurence.occurrenceData.basisOfRecord.slice(1),
        eventDate:                     format(occurence.occurrenceData.beginDate, 'YYYY-MM-DD') + '/' + format(occurence.occurrenceData.endDate, 'YYYY-MM-DD'),
        lifestage:                     occurence.occurrenceData.lifeStage,
        occurrenceStatus:              occurence.occurrenceData.occurrenceStatus,
        scientificName:                occurence.occurrenceData.scientificName,
        sex:                           occurence.occurrenceData.sex,
        decimalLongitude:              occurence.locationData.decimalLongitude,
        decimalLatitude:               occurence.locationData.decimalLatitude,
        coordinateUncertaintyInMeters: occurence.locationData.coordinateUncertainty,
        minimumDepthInMeters:          occurence.locationData.minimumDepth,
        maximumDepthInMeters:          occurence.locationData.maximumDepth,
        verbatimCoordinates:           occurence.locationData.verbatimCoordinates,
        verbatimDepth:                 occurence.locationData.verbatimDepth
      }
    }],
    emof:       [{
      purl:  {},
      iobis: {}
    }, {
      iobis: {}
    }]
  }
  const url = `/api/dwca/${occurence.dataset.id}/user/${userRef}/records`
  return await fetch(url, {
    method:  'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body:    JSON.stringify(occurrence)
  }).then(response => response.json())
}
