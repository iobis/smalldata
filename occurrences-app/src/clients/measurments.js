export function getGeneralMeasurements() {
  return [{
    type:   'Pressure',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/PRESPS02/',
    units:  [{ name: 'Kilogram', id: 'http://vocab.nerc.ac.uk/collection/P06/current/KGXX' }]
  }, {
    type:   'Temperature',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/TEMPCU01/',
    units:  [{ name: 'Degrees Celsius', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UPAA' }]
  }, {
    type:   'Salinity',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/PSALCU01/',
    units:  [{ name: 'Degrees Celsius', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UPAA' }]
  }]
}

export function getSpecificMeasurements() {
  return [{
    type:   'ObservedIndividualCount',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/OCOUNT01',
    units:  [{ name: 'Kilogram', id: 'http://vocab.nerc.ac.uk/collection/P06/current/KGXX' }]
  }, {
    type:   'Abundance category of biological entity specified elsewhere',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/SDBIOL06',
    units:  [{ name: 'Millimeter', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UXMM' }]
  }, {
    type:   'Abundance of biological entity specified elsewhere per unit area of the bed',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/SDBIOL02',
    units:  [{ name: 'Meter', id: 'http://vocab.nerc.ac.uk/collection/P06/current/ULAA/' }]
  }, {
    type:   'Abundance of biological entity specified elsewhere per unit volume of the sediment',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/USPBIOSV',
    units:  [{ name: 'Grams', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UGRM' }]
  }, {
    type:   'Abundance of biological entity specified elsewhere per unit volume of the water body',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/SDBIOL01',
    units:  [{ name: 'Micrometer', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UMIC' }]
  }, {
    type:   'Ash-free dry weight biomass (as carbon) of biological entity specified elsewhere per unit volume of the sediment',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/USPBICBM',
    units:  [{ name: 'Grams per square meter', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UGMS' }]
  }, {
    type:   'Ash-free dry weight biomass of biological entity specified elsewhere per unit area of the bed',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/SDBIOL03',
    units:  [{ name: 'Number per square meter', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UPMS' }]
  }, {
    type:   'Ash-free dry weight biomass of biological entity specified elsewhere per unit volume of the water body',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/SDBIOL07',
    units:  [{ name: 'Number per liter', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UCML' }]
  }, {
    type:   'Length of biological entity specified elsewhere',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/OBSINDLX/',
    units:  [{ name: 'Square meter', id: 'http://vocab.nerc.ac.uk/collection/P06/current/UMSQ' }]
  }, {
    type:   'Specimen weight of biological entity specified elsewhere',
    typeId: 'http://vocab.nerc.ac.uk/collection/P01/current/SPWGXX01/',
    units:  [{ name: 'Cubic metres', id: 'http://vocab.nerc.ac.uk/collection/P06/current/MCUB/' }]
  }]
}
