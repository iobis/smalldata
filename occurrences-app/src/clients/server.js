export function getOccurrenceMock() {
  return [{
    id:             1,
    addedDate:      '2011-12-01',
    scientificName: 'Abra Alba',
    dataset:        'NPPSD Short-tailed Albatross Sightings',
    occurrenceDate: '2011-12-09'
  }, {
    id:             2,
    addedDate:      '2011-12-01',
    scientificName: 'Abra Alba',
    dataset:        'NPPSD Short-tailed Albatross Sightings',
    occurrenceDate: '2011-12-09'
  }]
}
