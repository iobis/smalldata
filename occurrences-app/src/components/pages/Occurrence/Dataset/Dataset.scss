.dataset {
  cursor: pointer;
  padding-right: 1rem;
  padding-left: 1rem;

  .is-checkradio + label {
    padding-left: 0;
  }
}
