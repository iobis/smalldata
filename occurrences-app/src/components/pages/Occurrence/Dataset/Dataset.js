import InputRadio from '../../../form/InputRadio'
import PropTypes from 'prop-types'
import React from 'react'
import { datasetTitleOf } from '../../../../clients/SmalldataClient'
import ContinueButton from '../ContinueButton'

export default function Dataset({ datasets, selectedDataset, onChange, nextStepHandler }) {
  return (
    <>
      <div className="dataset is-fluid">
        <table className="table is-striped is-fullwidth">
          <tbody>
            {datasets.map(dataset => (
              <DatasetOption
                checked={dataset.id === selectedDataset.id}
                id={dataset.id}
                key={dataset.id}
                onClick={() => onChange(dataset)}
                title={datasetTitleOf(dataset)}/>))}
          </tbody>
        </table>
      </div>
      <ContinueButton
        name="datasetContinue"
        nextStepHandler={nextStepHandler}
        value="Continue to Occurrence Data"
        wrapperClassName=""/>
    </>
  )
}

export const datasetShape = {
  id:    PropTypes.string.isRequired,
  title: PropTypes.shape({
    value: PropTypes.string.isRequired
  }).isRequired
}

Dataset.propTypes = {
  datasets:        PropTypes.arrayOf(PropTypes.shape(datasetShape)).isRequired,
  nextStepHandler: PropTypes.func.isRequired,
  onChange:        PropTypes.func.isRequired,
  selectedDataset: PropTypes.shape(datasetShape).isRequired
}

function DatasetOption({ id, checked, title, onClick }) {
  return (
    <tr className="dataset-option" onClick={onClick}>
      <td><InputRadio checked={checked} name="dataset" onChange={onClick} value={id}/></td>
      <td>{title}</td>
    </tr>
  )
}

DatasetOption.propTypes = {
  checked: PropTypes.bool.isRequired,
  id:      PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  title:   PropTypes.string.isRequired
}
