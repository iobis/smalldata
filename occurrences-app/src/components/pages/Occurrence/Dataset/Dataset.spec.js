import React from 'react'
import Dataset from './Dataset'
import { RESPONSE_DEFAULT } from '../../../../clients/SmalldataClient.mock'
import { renameRefToId } from '../../../../clients/SmalldataClient'
import { mount } from 'enzyme'

describe('Dataset', () => {
  it('renders correctly', () => {
    expect(mount(
      <Dataset
        datasets={getDefaultDatasets()}
        onChange={() => {}}
        selectedDataset={getDefaultDatasets()[0]}/>
    )).toMatchSnapshot()
  })

  it('returns selected dataset when clicking on it', () => {
    const onChange = jest.fn()
    const wrapper = mount(
      <Dataset
        datasets={getDefaultDatasets()}
        onChange={onChange}
        selectedDataset={getDefaultDatasets()[0]}/>)

    wrapper.find('.dataset-option input').at(0).simulate('change')
    expect(onChange).toHaveBeenCalledTimes(1)
    expect(onChange).toHaveBeenNthCalledWith(1, getDefaultDatasets()[0])

    wrapper.find('.dataset-option input').at(1).simulate('change')
    expect(onChange).toHaveBeenCalledTimes(2)
    expect(onChange).toHaveBeenNthCalledWith(2, getDefaultDatasets()[1])

    wrapper.find('.dataset-option').at(2).simulate('click')
    expect(onChange).toHaveBeenCalledTimes(3)
    expect(onChange).toHaveBeenNthCalledWith(3, getDefaultDatasets()[2])
  })
})

function getDefaultDatasets() {
  return RESPONSE_DEFAULT.map(renameRefToId)
}
