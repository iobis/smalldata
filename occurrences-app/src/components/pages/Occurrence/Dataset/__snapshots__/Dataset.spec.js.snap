// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dataset renders correctly 1`] = `
<Dataset
  datasets={
    Array [
      Object {
        "abstract": Object {
          "paragraphs": Array [
            "Population structure data of Caprella species used in the article \\"Hosono T (2014) Temperature explains reproductive dynamics in caprellids at different latitudes. Marine Ecology Progress Series 511:129-141\\"",
          ],
        },
        "contacts": Array [
          Object {
            "individualName": Object {
              "givenName": "Takashi",
              "surName": "Hosono",
            },
          },
        ],
        "creators": Array [
          Object {
            "individualName": Object {
              "givenName": "Takashi",
              "surName": "Hosono",
            },
          },
        ],
        "id": "wEaBfmFyQhYCdsk",
        "keywordSets": Array [
          Object {
            "keywordThesaurus": "GBIF Dataset Type Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_type.xml",
            "keywords": Array [
              "Samplingevent",
            ],
          },
        ],
        "language": "en",
        "license": Object {
          "title": "Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License",
          "url": "http://creativecommons.org/licenses/by-nc/4.0/legalcode",
        },
        "meta": Object {
          "dwcTables": Object {
            "core": "event",
            "extensions": Array [
              "emof",
              "occurrence",
            ],
          },
          "type": "event",
        },
        "metadataProviders": Array [
          Object {
            "electronicMailAddress": "hosonot@jamstec.go.jp",
            "individualName": Object {
              "givenName": "Takashi",
              "surName": "Hosono",
            },
          },
        ],
        "title": Object {
          "language": "en",
          "value": "Caprellids polulation structure in Usujiri, Hokkaido, Japan",
        },
      },
      Object {
        "abstract": Object {
          "paragraphs": Array [
            "The data were collected in the Azov Sea during a cruise on board of R/V N. Danilevskiy in 1935. This dataset contains abundance data (individuals per m²) and biomass data (g/m²) for benthic taxa. No additional metadata is available.",
            "This dataset was rescued through data archaeology activities in the framework of the EMODnet Biology data grants.",
          ],
        },
        "contacts": Array [
          Object {
            "electronicMailAddress": "o.sergeyeva@gsva.net",
            "individualName": Object {
              "givenName": "Oleksandra",
              "surName": "Sergeyeva",
            },
            "organizationName": "National Academy of Sciences of Ukraine Institute of Biology of the Southern Seas (NASU-IBSS)",
          },
        ],
        "creators": Array [
          Object {
            "electronicMailAddress": "director@ibss.iuf.net",
            "individualName": Object {
              "givenName": "Yu. B.",
              "surName": "Belogurova",
            },
            "organizationName": "National Academy of Sciences of Ukraine Institute of Biology of the Southern Seas (NASU-IBSS)",
          },
          Object {
            "electronicMailAddress": "o.sergeyeva@gsva.net",
            "individualName": Object {
              "givenName": "Oleksandra",
              "surName": "Sergeyeva",
            },
            "organizationName": "National Academy of Sciences of Ukraine Institute of Biology of the Southern Seas (NASU-IBSS)",
          },
        ],
        "id": "ntDOtUc7XsRrIus",
        "keywordSets": Array [
          Object {
            "keywordThesaurus": "GBIF Dataset Type Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_type.xml",
            "keywords": Array [
              "Occurrence",
            ],
          },
          Object {
            "keywordThesaurus": "GBIF Dataset Subtype Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_subtype.xml",
            "keywords": Array [
              "Observation",
            ],
          },
        ],
        "language": "en",
        "license": Object {
          "title": "Creative Commons Attribution Non Commercial (CC-BY) 4.0 License",
          "url": "http://creativecommons.org/licenses/by/4.0/legalcode",
        },
        "meta": Object {
          "dwcTables": Object {
            "core": "occurrence",
            "extensions": Array [
              "emof",
            ],
          },
          "type": "occurrence",
        },
        "metadataProviders": Array [
          Object {
            "electronicMailAddress": "info@eurobis.org",
            "individualName": Object {
              "surName": "EurOBIS Data Management Team",
            },
            "organizationName": "Flanders Marine Institute (VLIZ)",
          },
        ],
        "title": Object {
          "language": "en",
          "value": "Benthos collected in the Azov Sea in 1935 on board the R/V \\"N. Danilevskiy\\"",
        },
      },
      Object {
        "abstract": Object {
          "paragraphs": Array [
            "An annotated check-list is given of Antipatharia species occurring deeper than 2000 m in the seas bordering Europe.",
            "The check-list is based on published data. It includes 12 species. For each species synonymy, data on localities in European seas and general species distribution are provided. Only distributions for which geographic coordinates are availlable were retained, as these are made available through EurOBIS.",
          ],
        },
        "contacts": Array [
          Object {
            "electronicMailAddress": "tina@ocean.ru",
            "individualName": Object {
              "givenName": "Tina",
              "surName": "Molodtsova",
            },
            "organizationName": "Russian Academy of Sciences P. P. Shirshov Institute of Oceanology (RAS-SIO RAS)",
          },
        ],
        "creators": Array [
          Object {
            "electronicMailAddress": "tina@ocean.ru",
            "individualName": Object {
              "givenName": "Tina",
              "surName": "Molodtsova",
            },
            "organizationName": "Russian Academy of Sciences P. P. Shirshov Institute of Oceanology (RAS-SIO RAS)",
          },
        ],
        "id": "PoJnGNMaxsupE4w",
        "keywordSets": Array [
          Object {
            "keywordThesaurus": "GBIF Dataset Type Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_type.xml",
            "keywords": Array [
              "Occurrence",
            ],
          },
          Object {
            "keywordThesaurus": "GBIF Dataset Subtype Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_subtype.xml",
            "keywords": Array [
              "Observation",
            ],
          },
        ],
        "language": "en",
        "license": Object {
          "title": "Creative Commons Attribution Non Commercial (CC-BY) 4.0 License",
          "url": "http://creativecommons.org/licenses/by/4.0/legalcode",
        },
        "meta": Object {
          "dwcTables": Object {
            "core": "occurrence",
            "extensions": Array [
              "emof",
            ],
          },
          "type": "occurrence",
        },
        "metadataProviders": Array [
          Object {
            "electronicMailAddress": "info@eurobis.org",
            "individualName": Object {
              "surName": "EurOBIS Data Management Team",
            },
            "organizationName": "Flanders Marine Institute (VLIZ)",
          },
        ],
        "title": Object {
          "language": "en",
          "value": "Antipatharia distribution data from: Deep-sea fauna of European seas - an annotated species check-list of benthic invertebrates living deeper than 2000 m in the seas bordering Europe",
        },
      },
      Object {
        "abstract": Object {
          "paragraphs": Array [
            "The dataset contains the abundance and biomass values of sediment organisms from the Sevastopol region (South-Western Crimea, The Black Sea, Ukraine) collected between 1982 and 1992.",
            "Abundance and Biomass of benthic organisms and key abiotic variables (covering irregular sampling surveys within 1982-1992) from the Sevastopol Area (South-Western Crimea, The Black Sea, Ukraine) including estuaries and small sheltered and semi-closed bays within the area of &quot;big Sevastopol bay&quot;
Sevastopol Bay is located in south western part of the Crimean peninsula, Ukraine and comprehends the one of the biggest marine harbours of the Black Sea and has served as an industrialized naval base for decades. A consequence of this long-term industrial, military and municipal activity was the deterioration of the marine environment of the bay, depressed state of benthic assemblages and occurrence of expanded bottom areas that became contaminated with various sediment-associated pollutants including polycyclic aromatic hydrocarbons (PAHs), polychlorinated biphenyls (PCBs), pesticides, and metals. Benthic samples in Sevastopol bay heve been taken between 1982 and 1992.",
          ],
        },
        "contacts": Array [
          Object {
            "electronicMailAddress": "arvanitidis@her.hcmr.gr",
            "individualName": Object {
              "givenName": "Christos",
              "surName": "Arvanitidis",
            },
            "organizationName": "Hellenic Centre for Marine Research (HCMR)",
          },
        ],
        "creators": Array [
          Object {
            "electronicMailAddress": "arvanitidis@her.hcmr.gr",
            "individualName": Object {
              "givenName": "Christos",
              "surName": "Arvanitidis",
            },
            "organizationName": "Hellenic Centre for Marine Research (HCMR)",
          },
          Object {
            "organizationName": "Hellenic Centre for Marine Research (HCMR)",
          },
          Object {
            "electronicMailAddress": "alexpet@ibss.iuf.net",
            "individualName": Object {
              "givenName": "Alexei",
              "surName": "Petrov",
            },
            "organizationName": "National Academy of Sciences of Ukraine Institute of Biology of the Southern Seas (NASU-IBSS",
          },
        ],
        "id": "NnqVLwIyPn-nRkc",
        "keywordSets": Array [
          Object {
            "keywordThesaurus": "GBIF Dataset Type Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_type.xml",
            "keywords": Array [
              "Occurrence",
            ],
          },
          Object {
            "keywordThesaurus": "GBIF Dataset Subtype Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_subtype.xml",
            "keywords": Array [
              "Observation",
            ],
          },
          Object {
            "keywordThesaurus": "ASFA",
            "keywords": Array [
              "Benthic biomass",
              "Benthos",
              "Data",
              "Marine Genomics",
            ],
          },
        ],
        "language": "en",
        "license": Object {
          "title": "Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License",
          "url": "http://creativecommons.org/licenses/by-nc/4.0/legalcode",
        },
        "meta": Object {
          "dwcTables": Object {
            "core": "occurrence",
            "extensions": Array [
              "emof",
            ],
          },
          "type": "occurrence",
        },
        "metadataProviders": Array [
          Object {
            "electronicMailAddress": "info@eurobis.org",
            "individualName": Object {
              "surName": "EurOBIS Data Management Team",
            },
            "organizationName": "Flanders Marine Institute (VLIZ)",
          },
        ],
        "title": Object {
          "language": "en",
          "value": "Benthic data from Sevastopol (Black Sea)",
        },
      },
    ]
  }
  nextStepHandler={[Function]}
  onChange={[Function]}
  selectedDataset={
    Object {
      "abstract": Object {
        "paragraphs": Array [
          "Population structure data of Caprella species used in the article \\"Hosono T (2014) Temperature explains reproductive dynamics in caprellids at different latitudes. Marine Ecology Progress Series 511:129-141\\"",
        ],
      },
      "contacts": Array [
        Object {
          "individualName": Object {
            "givenName": "Takashi",
            "surName": "Hosono",
          },
        },
      ],
      "creators": Array [
        Object {
          "individualName": Object {
            "givenName": "Takashi",
            "surName": "Hosono",
          },
        },
      ],
      "id": "wEaBfmFyQhYCdsk",
      "keywordSets": Array [
        Object {
          "keywordThesaurus": "GBIF Dataset Type Vocabulary: http://rs.gbif.org/vocabulary/gbif/dataset_type.xml",
          "keywords": Array [
            "Samplingevent",
          ],
        },
      ],
      "language": "en",
      "license": Object {
        "title": "Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License",
        "url": "http://creativecommons.org/licenses/by-nc/4.0/legalcode",
      },
      "meta": Object {
        "dwcTables": Object {
          "core": "event",
          "extensions": Array [
            "emof",
            "occurrence",
          ],
        },
        "type": "event",
      },
      "metadataProviders": Array [
        Object {
          "electronicMailAddress": "hosonot@jamstec.go.jp",
          "individualName": Object {
            "givenName": "Takashi",
            "surName": "Hosono",
          },
        },
      ],
      "title": Object {
        "language": "en",
        "value": "Caprellids polulation structure in Usujiri, Hokkaido, Japan",
      },
    }
  }
>
  <div
    className="dataset is-fluid"
  >
    <table
      className="table is-striped is-fullwidth"
    >
      <tbody>
        <DatasetOption
          checked={true}
          id="wEaBfmFyQhYCdsk"
          key="wEaBfmFyQhYCdsk"
          onClick={[Function]}
          title="Caprellids polulation structure in Usujiri, Hokkaido, Japan"
        >
          <tr
            className="dataset-option"
            onClick={[Function]}
          >
            <td>
              <InputRadio
                checked={true}
                name="dataset"
                onChange={[Function]}
                value="wEaBfmFyQhYCdsk"
              >
                <input
                  checked={true}
                  className="input-radio is-checkradio"
                  id="dataset-wEaBfmFyQhYCdsk"
                  name="dataset"
                  onChange={[Function]}
                  type="radio"
                  value="wEaBfmFyQhYCdsk"
                />
                <label
                  htmlFor="dataset-wEaBfmFyQhYCdsk"
                />
              </InputRadio>
            </td>
            <td>
              Caprellids polulation structure in Usujiri, Hokkaido, Japan
            </td>
          </tr>
        </DatasetOption>
        <DatasetOption
          checked={false}
          id="ntDOtUc7XsRrIus"
          key="ntDOtUc7XsRrIus"
          onClick={[Function]}
          title="Benthos collected in the Azov Sea in 1935 on board the R/V \\"N. Danilevskiy\\""
        >
          <tr
            className="dataset-option"
            onClick={[Function]}
          >
            <td>
              <InputRadio
                checked={false}
                name="dataset"
                onChange={[Function]}
                value="ntDOtUc7XsRrIus"
              >
                <input
                  checked={false}
                  className="input-radio is-checkradio"
                  id="dataset-ntDOtUc7XsRrIus"
                  name="dataset"
                  onChange={[Function]}
                  type="radio"
                  value="ntDOtUc7XsRrIus"
                />
                <label
                  htmlFor="dataset-ntDOtUc7XsRrIus"
                />
              </InputRadio>
            </td>
            <td>
              Benthos collected in the Azov Sea in 1935 on board the R/V "N. Danilevskiy"
            </td>
          </tr>
        </DatasetOption>
        <DatasetOption
          checked={false}
          id="PoJnGNMaxsupE4w"
          key="PoJnGNMaxsupE4w"
          onClick={[Function]}
          title="Antipatharia distribution data from: Deep-sea fauna of European seas - an annotated species check-list of benthic invertebrates living deeper than 2000 m in the seas bordering Europe"
        >
          <tr
            className="dataset-option"
            onClick={[Function]}
          >
            <td>
              <InputRadio
                checked={false}
                name="dataset"
                onChange={[Function]}
                value="PoJnGNMaxsupE4w"
              >
                <input
                  checked={false}
                  className="input-radio is-checkradio"
                  id="dataset-PoJnGNMaxsupE4w"
                  name="dataset"
                  onChange={[Function]}
                  type="radio"
                  value="PoJnGNMaxsupE4w"
                />
                <label
                  htmlFor="dataset-PoJnGNMaxsupE4w"
                />
              </InputRadio>
            </td>
            <td>
              Antipatharia distribution data from: Deep-sea fauna of European seas - an annotated species check-list of benthic invertebrates living deeper than 2000 m in the seas bordering Europe
            </td>
          </tr>
        </DatasetOption>
        <DatasetOption
          checked={false}
          id="NnqVLwIyPn-nRkc"
          key="NnqVLwIyPn-nRkc"
          onClick={[Function]}
          title="Benthic data from Sevastopol (Black Sea)"
        >
          <tr
            className="dataset-option"
            onClick={[Function]}
          >
            <td>
              <InputRadio
                checked={false}
                name="dataset"
                onChange={[Function]}
                value="NnqVLwIyPn-nRkc"
              >
                <input
                  checked={false}
                  className="input-radio is-checkradio"
                  id="dataset-NnqVLwIyPn-nRkc"
                  name="dataset"
                  onChange={[Function]}
                  type="radio"
                  value="NnqVLwIyPn-nRkc"
                />
                <label
                  htmlFor="dataset-NnqVLwIyPn-nRkc"
                />
              </InputRadio>
            </td>
            <td>
              Benthic data from Sevastopol (Black Sea)
            </td>
          </tr>
        </DatasetOption>
      </tbody>
    </table>
  </div>
  <ContinueButton
    name="datasetContinue"
    nextStepHandler={[Function]}
    value="Continue to Occurrence Data"
    wrapperClassName=""
  >
    <div
      className=""
    >
      <button
        className="button is-info"
        name="datasetContinue"
        onClick={[Function]}
      >
        Continue to Occurrence Data
      </button>
    </div>
  </ContinueButton>
</Dataset>
`;
